src/CaptureFile/CaptureFileOutputStream.cpp

```diff
@@ -66,14 +68,26 @@ ErrorMessageOr<void> CaptureFileOutputStreamImpl::Initialize() {
   return outcome::success();
 }
 
-void CaptureFileOutputStreamImpl::Close() noexcept {
+ErrorMessageOr<void> CaptureFileOutputStreamImpl::Close() noexcept {
+  coded_output_->Trim();
+  if (coded_output_->HadError()) {
+    return HandleWriteError("Unknown", SafeStrerror(file_output_stream_->GetErrno()));
```

Comment:
Don't we need also need to close the streams in case of an error?

Reply:
HandleWriteError resets the stream and file and also removes it.

Reply:
Okay, thanks.